programs/account-compression/tests/test.rs

```diff
@@ -0,0 +1,394 @@
+            &context.payer.pubkey(),
+            &merkle_tree_pubkey,
+            context
+                .banks_client
+                .get_rent()
+                .await
+                .unwrap()
+                .minimum_balance(account_compression::ConcurrentMerkleTreeAccount::LEN),
+            account_compression::ConcurrentMerkleTreeAccount::LEN as u64,
+            &account_compression::ID,
+        );
+
+        let instruction = Instruction {
+            program_id: account_compression::ID,
+            accounts: vec![
+                AccountMeta::new(context.payer.pubkey(), true),
+                AccountMeta::new(merkle_tree_pubkey, false),
+                AccountMeta::new_readonly(system_program::ID, false),
+            ],
+            data: instruction_data.data(),
+        };
+
+        let transaction = Transaction::new_signed_with_payer(
+            &[account_create_ix, instruction],
+            Some(&context.payer.pubkey()),
+            &vec![&context.payer, &merkle_tree_keypair],
+            context.last_blockhash,
+        );
+        context
+            .banks_client
+            .process_transaction(transaction.clone())
+            .await
+            .unwrap();
+        let merkle_tree =
+            get_account_zero_copy::<account_compression::ConcurrentMerkleTreeAccount>(
+                &mut context,
+                merkle_tree_pubkey,
+            )
+            .await;
+        assert_eq!(merkle_tree.owner, context_pubkey);
+        assert_eq!(merkle_tree.delegate, context_pubkey);
+        assert_eq!(merkle_tree.index, 1);
+
+        // insertions with merkle tree leaves missmatch should fail
+        let instruction_data = account_compression::instruction::InsertLeavesIntoMerkleTrees {
+            leaves: vec![[1u8; 32], [2u8; 32]],
+        };
+
+        let accounts = account_compression::accounts::InsertTwoLeavesParallel {
+            authority: context.payer.pubkey(),
+            registered_verifier_pda: None,
+            log_wrapper: account_compression::state::event::NOOP_PROGRAM_ID,
+        };
+        // accounts.
+
+        let instruction = Instruction {
+            program_id: account_compression::ID,
+            accounts: vec![
+                accounts.to_account_metas(Some(true)),
+                vec![
+                    AccountMeta::new(merkle_tree_pubkey, false),
+                    // AccountMeta::new(merkle_tree_pubkey, false),
+                ],
+            ]
+            .concat(),
+            data: instruction_data.data(),
+        };
+
+        let transaction = Transaction::new_signed_with_payer(
+            &[instruction],
+            Some(&context.payer.pubkey()),
+            &vec![&context.payer],
+            context.last_blockhash,
+        );
+        let remaining_accounts_missmatch_error =
+            context.banks_client.process_transaction(transaction).await;
+        assert!(remaining_accounts_missmatch_error.is_err());
```

Comment:
You can use `matches!(remaining_accounts_mismatch_error, ErrorCode::Something)` if you want to check an exact error type.